src/RatcliffDDM.jl

```diff
@@ -1,371 +1,605 @@
+           
+        if isodd(i)
+            S += 4 * y
+        else
+            S += 2 * y
+        end
+
+    end
     
-#     S = S - y  # the last term should be f(b) and not 2*f(b) so we subtract y
-#     S = S / ((ub_t - lb_t) + (ub_z - lb_z))  # the right function if pdf_sv()/sz or pdf_sv()/st
+    S = S - y  # the last term should be f(b) and not 2*f(b) so we subtract y
+    S = S / ((ub_t - lb_t) + (ub_z - lb_z))  # the right function if pdf_sv()/sz or pdf_sv()/st
     
-#     return (ht + hz) * S / 3
-
-# end
-
-# """
-#     RatcliffDDM 
-
-#     Model object for the Ratcliff Diffusion Model.
-
-# # Fields
-#     - `ν`: drift rate. Average slope of the information accumulation process. The drift gives information about the speed and direction of the accumulation of information. Typical range: -5 < ν < 5
-#     - `α`: boundary threshold separation. The amount of information that is considered for a decision. Typical range: 0.5 < α < 2
-#     - `τ`: non-decision time. The duration for a non-decisional processes (encoding and response execution). Typical range: 0.1 < τ < 0.5 
-#     - `z`: starting point. Indicator of an an initial bias towards a decision. The z parameter is relative to a (i.e. it ranges from 0 to 1).
-#     - `η`:  across-trial-variability of drift rate. Typical range: 0 < η < 2. Default is 0.
-#     - `sz`: across-trial-variability of starting point. Typical range: 0 < sz < 0.5. Default is 0.
-#     - `st`: across-trial-variability of non-decision time. Typical range: 0 < st < 0.2. Default is 0.
-#     - `σ`: diffusion noise constant. Default is 1.
-
-# # Example 
-
-# ````julia
-# using SequentialSamplingModels
-# dist = RatcliffDDM(ν = 0.50,α = 0.08,τ = 0.30,z = 0.04,η = 0.10,sz = 0.02,st = .02,σ = 0.10) 
-# choice,rt = rand(dist, 10)
-# like = pdf.(dist, choice, rt)
-# loglike = logpdf.(dist, choice, rt)
-# ````
+    return (ht + hz) * S / 3
+
+end
+
+# Simpson's Method two dimentional case
+function _simpson_2D(x::Real, ν::Real, η::Real, α::Real, z::Real, τ::Real, ϵ::Real, lb_z::Real, ub_z::Real, n_sz::Int, lb_t::Real, ub_t::Real, n_st::Int)
+ #   @assert (n_sz & 1) == 0 && (n_st & 1) == 0     # n_st and n_sz have to be even
+ #   @assert (ub_t-lb_t)*(ub_z-lb_z)>0 && (n_sz*n_st)>0  # 2D-integration only
+
+    ht = (ub_t-lb_t)/n_st
+    S = _simpson_1D(x, ν, η, α, z, τ, ϵ, lb_z, ub_z, n_sz, 0, 0, 0)
+
+    t_tag = 0
+    y = 0 
+    for i_t in 1:n_st
+        t_tag = lb_t + ht * i_t
+        y = _simpson_1D(x, ν, η, α, z, t_tag, ϵ, lb_z, ub_z, n_sz, 0, 0, 0)
+
+        if isodd(i_t)
+            S += 4 * y
+        else
+            S += 2 * y
+        end
+    end
+
+    S = S - y  # the last term should be f(b) and not 2*f(b) so we subtract y
+    S = S / (ub_t - lb_t)
+
+    return ht * S / 3
+
+end
+
+logpdf(d::RatcliffDDM, choice, rt; ϵ::Real = 1.0e-12) = log(pdf(d, choice, rt; ϵ))
+
+function logpdf(d::RatcliffDDM, data::T) where {T<:NamedTuple}
```

Comment:
I think this function is missing `ϵ::Real = 1.0e-12`

Reply:
Actually, I think you can delete this method if you modify this generic [method](https://github.com/itsdfish/SequentialSamplingModels.jl/blob/7f9151d504ef6a1163154927d5515da8a9ff684c/src/utilities.jl#L51) to be 

`logpdf(d::SSM2D, data::NamedTuple; kwargs...) = logpdf.(d, data.choice, data.rt;  kwargs...)`

When the method above is called, it will pass the inputs to `logpdf(d::RatcliffDDM, choice, rt; ϵ::Real = 1.0e-12)` where the default value of `ϵ` will be used if not keyword arguments are passed.